pkg/channel/producer/producer.go

```diff
@@ -0,0 +1,161 @@
+package producer
+
+import (
+	"errors"
+	"github.com/cloudevents/sdk-go/v1/cloudevents"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"knative.dev/eventing-kafka/pkg/channel/health"
+	"knative.dev/eventing-kafka/pkg/channel/message"
+	"knative.dev/eventing-kafka/pkg/channel/util"
+	kafkaproducer "knative.dev/eventing-kafka/pkg/common/kafka/producer"
+	"knative.dev/eventing-kafka/pkg/common/prometheus"
+	"go.uber.org/zap"
+	eventingChannel "knative.dev/eventing/pkg/channel"
+)
+
+// Package Variables
+var (
```

Comment:
just curious why this is not a struct / instance for each producer.

Reply:
No reason really - we only currently have a single producer so it was implemented as a pkg level singleton.  You're correct that a typed/struct for the producer would be more standard - I can convert the implementation to that pretty easily.